### ISSUE TITLE
RTS / Save States

### ISSUE THREAD

**Issue by ahezard** [ahezard/nds-bootstrap#143]

**Comment by Anonymous941** [comment]
I thought this was impossible on hypervisors because lack of direct hardware access?

**Comment by Peter0x44** [comment]
There are flashcards which can do this. I think it's tricky but possible, no work or research has been done for it yet though.

**Comment by CamperSamu** [comment]
The savestate branch hasn't been updated in years, I suppose the research done has led to negative results?

Dumb question from someone with little knowledge about the system and the project: since NDS-Bootstrap right now has RAM dumping and writing (at least on 3DS), shouldn't it be possible by just writing a previous dump to memory?

**Comment by Anonymous941** [comment]
> Dumb question from someone with little knowledge about the system and the project: since NDS-Bootstrap right now has RAM dumping and writing (at least on 3DS), shouldn't it be possible by just writing a previous dump to memory?

No, you have to restore the state perfectly for all of the various hardware components, not just RAM.  afaik there's no other console that you can save and load state on original hardware, even if it's just backwards compatibility, and I don't understand how it's possible but I guess the NDS must be unique somehow

**Comment by Invictaz** [comment]
> > Dumb question from someone with little knowledge about the system and the project: since NDS-Bootstrap right now has RAM dumping and writing (at least on 3DS), shouldn't it be possible by just writing a previous dump to memory?
> 
> No, you have to restore the state perfectly for all of the various hardware components, not just RAM. afaik there's no other console that you can save and load state on original hardware, even if it's just backwards compatibility, and I don't understand how it's possible but I guess the NDS must be unique somehow

Incorrect. There is one plugin I can think of which is https://github.com/TheOfficialFloW/dejavu for Playstation Vita.

**Comment by coderkei** [comment]
> > Dumb question from someone with little knowledge about the system and the project: since NDS-Bootstrap right now has RAM dumping and writing (at least on 3DS), shouldn't it be possible by just writing a previous dump to memory?
> 
> No, you have to restore the state perfectly for all of the various hardware components, not just RAM. afaik there's no other console that you can save and load state on original hardware, even if it's just backwards compatibility, and I don't understand how it's possible but I guess the NDS must be unique somehow

The DS also has many write-only registers which can't be read which is why Real Time Save on most flashcarts cause some games to just crash. Its never been a reliable feature on any flashcart, even on the Supercard DSTWO which is regarded as having the best RTS. It makes no difference for nds-bootstrap because we are dealing with the same limitations, and its still largely a gimmick in every implementation I have seen.

**Comment by Anonymous941** [comment]
> Incorrect. There is one plugin I can think of which is https://github.com/TheOfficialFloW/dejavu for Playstation Vita.

Interesting, I didn't know about that (also that's why I said afaik ;))
